process/sync/metablock.go

```diff
@@ -376,6 +377,17 @@ func (boot *MetaBootstrap) doJobOnSyncBlockFail(hdr *block.MetaBlock, err error)
 			log.Info(errNotCritical.Error())
 		}
 	}
+
+	if err == process.ErrInvalidBlockHash {
```

Comment:
what is this doing ?

Reply:
Actually this will fix a situation when all the nodes would been replaced in their headerNonceHash pool an used header in their blockchain construction, with a wrong/unused block header on which they didn't construct, coming after a late broadcast from a valid proposer, and because of this, all the new nodes which are bootstrapping would be stuck and would not advance from that point. This mechanism could be backed up by some additional information which could be send to the interceptors and in this way they would avoid replacing good data with these kind of unused/old coming stuff.

Reply:
put this as a hot fix PR into the master.